sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/LookupCatalog.scala

```diff
@@ -26,27 +28,31 @@ import org.apache.spark.sql.catalyst.TableIdentifier
 @Experimental
 trait LookupCatalog {
 
-  def lookupCatalog: Option[(String) => CatalogPlugin] = None
+  def lookupCatalog: Option[String => CatalogPlugin] = None
 
   type CatalogObjectIdentifier = (Option[CatalogPlugin], Identifier)
 
   /**
    * Extract catalog plugin and identifier from a multi-part identifier.
    */
   object CatalogObjectIdentifier {
-    def unapply(parts: Seq[String]): Option[CatalogObjectIdentifier] = lookupCatalog.map { lookup =>
-      parts match {
-        case Seq(name) =>
-          (None, Identifier.of(Array.empty, name))
-        case Seq(catalogName, tail @ _*) =>
-          try {
-            val catalog = lookup(catalogName)
-            (Some(catalog), Identifier.of(tail.init.toArray, tail.last))
-          } catch {
-            case _: CatalogNotFoundException =>
-              (None, Identifier.of(parts.init.toArray, parts.last))
-          }
-      }
+    def unapply(parts: Seq[String]): Option[CatalogObjectIdentifier] = parts match {
```

Comment:
@jzhuge, can we return `Some` instance?

Reply:
Thanks @HyukjinKwon for the review.
Not quite sure what you meant. Method unapply has to return Option[T].

Reply:
I meant to change return type to `Some[CatalogObjectIdentifier]` becuase `unapply` can specify `Some[T]` too.

```scala
scala> object A {
     |   def unapply(arg: String): Some[String] = Some("a")
     | }
defined object A

scala> val A(a) = "aaa"
a: String = a

scala> a
res1: String = a

scala> "aaa" match {
     |     case A(a) => println(a)
     | }
a
```

Reply:
I think the intent is to signal that `unapply` always works, right?

Reply:
Ah, nice! Could not find it in https://www.scala-lang.org/files/archive/spec/2.11/08-pattern-matching.html#extractor-patterns. Is it documented somewhere?
Tried it and IntelliJ did give warning when I attempted to return None when the return type is Some[T]! 
<img width="702" alt="Screen Shot 2019-05-27 at 4 21 23 PM" src="https://user-images.githubusercontent.com/1883812/58441990-a2230100-809b-11e9-8395-090208232160.png">